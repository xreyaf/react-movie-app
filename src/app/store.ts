import { configureStore } from "@reduxjs/toolkit";
import movieApi from "../features/movies/movieApi";

export const store = configureStore({
  reducer: {
    [movieApi.reducerPath]: movieApi.reducer
  },
  middleware: (getDefaultMiddleware) => {
    return getDefaultMiddleware().concat(movieApi.middleware);
  }
});
export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
