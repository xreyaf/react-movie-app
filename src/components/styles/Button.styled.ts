import styled from "@emotion/styled";
import { theme } from "../../styles/theme";
import { motion } from "framer-motion";

export const ButtonStyled = styled(motion.button)`
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 0.5rem;
  outline: none;
  border: none;
  border-radius: 1rem;
  background: ${theme.colors.primary400};
  color: ${theme.colors.white100};
  padding: 1rem 2rem;
  font-size: 1rem;
  margin-bottom: .6rem;
  svg{
    align-self: baseline;
  }
`;
