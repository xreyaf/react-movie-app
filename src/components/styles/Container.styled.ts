import styled from "@emotion/styled";

export const Container = styled.div`
  width: 100%;
  max-width: 1440px;
  padding: 1rem 5rem;
  margin: 0 auto;
`;
