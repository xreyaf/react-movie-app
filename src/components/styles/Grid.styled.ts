import styled from "@emotion/styled";
import { motion } from "framer-motion";

export const GridStyled = styled(motion.div)`
  display: grid;
  grid-template-columns: repeat(4, minmax(0, 1fr));
  row-gap: 1.4rem;
  column-gap: 1.4rem;
  align-items: start;
  @media screen and (max-width: 1199.98px) {
    grid-template-columns: repeat(3, minmax(0, 1fr));
    transition: all .5s;
  }
  @media screen and (max-width: 991.98px) {
    row-gap: .8rem;
    column-gap: .8rem;
    transition: all .5s;
  }
  @media screen and (max-width: 816.98px) {
    grid-template-columns: repeat(2, minmax(0, 1fr));
    row-gap: .8rem;
    column-gap: .8rem;
    transition: all .5s;
  }
  @media screen and (max-width: 575.98px) {
    grid-template-columns: repeat(1, minmax(0, 1fr));
    transition: all .5s;
  }
`;
