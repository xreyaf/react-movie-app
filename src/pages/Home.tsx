import React, { ChangeEvent, useCallback, useState } from 'react';
// @ts-ignore
import * as Unicons from '@iconscout/react-unicons';
import Card from '../components/Card';
import {
  useGetTrendingQuery,
  useSearchMediaQuery,
} from '../features/movies/TMDBApi';
import debounce from 'lodash/debounce';

import Spinner from '../components/Spinner';
import Input from '../components/Input';
import Grid from '../components/Grid';
import Container from '../components/Container';
import Heading from '../components/Heading';
import { Space } from '../components/layouts/Space';

const Home = () => {
  const [searchTitle, setSearchTitle] = useState('');
  const mediaType = 'all';
  const time = 'week';

  const {
    data: trending,
    isLoading: isLoadingTrending,
    error: errorTrending,
  } = useGetTrendingQuery({
    mediaType,
    time,
  });
  const {
    data: search,
    isLoading: isLoadingSearch,
    error: errorSearch,
  } = useSearchMediaQuery(searchTitle);

  const debouncedSearch = useCallback(
    debounce((q) => setSearchTitle(q), 600),
    []
  );

  const onChangeInput = (e: ChangeEvent<HTMLInputElement>) => {
    debouncedSearch(e.target.value);
  };

  let data;
  searchTitle ? (data = search) : (data = trending);

  return (
    <Container>
      <Heading>Сейчас в тренде</Heading>
      <Input
        name={'search'}
        onChange={onChangeInput}
        placeholder="напр. Брат"
        label="Найти фильм или сериал"
        type="text"
        icon={<Unicons.UilSearch size={24} />}
      />
      {isLoadingTrending && isLoadingSearch && (
        <>
          <Spinner />
          <Space />
        </>
      )}
      {errorTrending && errorSearch && (
        <>
          <Heading>Произошла ошибка...</Heading>
          <Space />
        </>
      )}
      {data && data.length != 0 ? (
        <>
          <Grid>
            {data.map((media) => (
              <Card key={media.id} type={media.media_type} item={media} />
            ))}
          </Grid>
        </>
      ) : (
        <>
          <Heading>Не нашлось результатов :(</Heading>
          <Space />
        </>
      )}
    </Container>
  );
};

export default Home;
