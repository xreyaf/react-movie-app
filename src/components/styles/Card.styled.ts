import styled from "@emotion/styled";
import { theme } from "../../styles/theme";
import { motion } from "framer-motion";

export const CardStyled = styled(motion.div)`
  position: relative;
  border-radius: 1rem;
  overflow: hidden;
  background-color: ${theme.colors.grey900_90};
  width: 100%;
  min-height: 100%;
`;
export const CardImageWrapper = styled.div`
  display: inline-block;
  border-radius: 1rem;
  width: 100%;
  height: 88%;
  overflow: hidden;
`;

export const ImageStyled = styled(motion.img)`
  object-fit: cover;
  width: 100%;
  min-height: 100%;
  vertical-align: middle;
`;

export const CardInfo = styled.div`
  padding: .5rem;
  margin-bottom: .5rem;
  overflow: hidden;

  a {
    color: ${theme.colors.grey50};
    font-weight: 600;

    &:hover {
      color: ${theme.colors.white100};
      transition: color .3s;
    }
  }
`;