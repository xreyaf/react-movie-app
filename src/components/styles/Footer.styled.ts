import styled from "@emotion/styled";
import { theme } from "../../styles/theme";

export const FooterStyled = styled.footer`
  background: ${theme.colors.grey900_80};
  backdrop-filter: blur(2px);
  padding: 5rem 0;
  text-align: center;
`;
