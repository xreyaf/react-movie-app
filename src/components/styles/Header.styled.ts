import styled from "@emotion/styled";
import { theme } from "../../styles/theme";
import { motion } from "framer-motion";

export const HeaderStyled = styled.header`
  position: fixed;
width: 100%;
height: 10%;
  top: 0;
  left: 0;


  z-index: 100;
  background: ${theme.colors.grey900_80};
  backdrop-filter: blur(24px);

  

`;

export const Nav = styled.nav`
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 3rem;

  ul {
    display: flex;
    flex-direction: row;
    justify-content: end;
    align-items: center;
  }
`;

export const Logo = styled(motion.img)`
  max-width: 3rem;
`;

export const StyledLinkMenu = styled(motion.a)`
  display: inline-flex;
  gap: 0.5rem;
  align-items: center;
  padding: 1rem 1rem;
  font-weight: 700;
  font-size: 1.2rem;
  cursor: pointer;
  color: ${theme.colors.grey400};

  &:hover {
    color: ${theme.colors.primary400};
  }
`;
