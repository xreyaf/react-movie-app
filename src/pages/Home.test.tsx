import { theme } from '../styles/theme';
import Home from './Home';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../app/store';
import { ThemeProvider } from '@emotion/react';

test('it works', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <Home />
      </Provider>
    </ThemeProvider>
  );
  expect(container.firstChild).toMatchSnapshot();
});
