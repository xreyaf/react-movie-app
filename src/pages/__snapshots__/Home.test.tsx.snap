// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.emotion-0 {
  width: 100%;
  max-width: 1440px;
  padding: 1rem 5rem;
  margin: 0 auto 2rem auto;
  height: 100%;
}

@media screen and (max-width: 1199.98px) {
  .emotion-0 {
    padding: 1rem 2rem;
  }
}

@media screen and (max-width: 991.98px) {
  .emotion-0 {
    padding: 1rem 1.5rem;
  }
}

@media screen and (max-width: 816.98px) {
  .emotion-0 {
    padding: 1rem 1.2rem;
  }
}

@media screen and (max-width: 575.98px) {
  .emotion-0 {
    padding: 1rem 1rem;
  }
}

.emotion-1 {
  position: relative;
  border-radius: 1rem;
  margin-bottom: 3rem;
}

.emotion-1 label {
  position: absolute;
  top: 1.75rem;
  left: 4rem;
  font-size: 1.3rem;
  color: hsla(224, 19.3%, 34.5%, 1);
  cursor: text;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 0.3s cubic-bezier(0.165, 0.84, 0.44, 1);
  transition: all 0.3s cubic-bezier(0.165, 0.84, 0.44, 1);
}

.emotion-2 {
  display: inline-block;
  position: absolute;
  top: 1.8rem;
  left: 1.5rem;
  color: hsla(224, 19.3%, 34.5%, 1);
}

.emotion-3 {
  background: transparent;
  background: hsla(0, 0%, 0%, 0.1);
  z-index: 3;
  width: 100%;
  height: 5rem;
  padding: 1rem 1.3rem 0.3rem 3.9rem;
  font-size: 1.3rem;
  border-radius: 1rem;
  color: hsla(224, 12.3%, 52.2%, 1);
  border: 3px solid hsla(224, 31.9%, 18.4%, 1);
  outline: none;
}

.emotion-3:focus {
  border: 3px solid hsla(246, 88.5%, 62.4%, 1);
}

.emotion-3:focus+label {
  -webkit-transform: translateY(-1.2rem);
  -moz-transform: translateY(-1.2rem);
  -ms-transform: translateY(-1.2rem);
  transform: translateY(-1.2rem);
  font-size: 1rem;
}

.emotion-3:focus::-webkit-input-placeholder {
  opacity: 0.5;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
}

.emotion-3:focus::-moz-placeholder {
  opacity: 0.5;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
}

.emotion-3:focus:-ms-input-placeholder {
  opacity: 0.5;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
}

.emotion-3:focus::placeholder {
  opacity: 0.5;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
}

.emotion-3::-webkit-input-placeholder {
  opacity: 0;
}

.emotion-3::-moz-placeholder {
  opacity: 0;
}

.emotion-3:-ms-input-placeholder {
  opacity: 0;
}

.emotion-3::placeholder {
  opacity: 0;
}

.emotion-3:not(:placeholder-shown)+label {
  -webkit-transform: translateY(-1.2rem);
  -moz-transform: translateY(-1.2rem);
  -ms-transform: translateY(-1.2rem);
  transform: translateY(-1.2rem);
  font-size: 1rem;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 3rem 0;
}

<div
  class="emotion-0"
>
  <h1>
    Trending now
  </h1>
  <div
    class="emotion-1"
  >
    <div
      class="emotion-2"
    >
      <svg
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21.71,20.29,18,16.61A9,9,0,1,0,16.61,18l3.68,3.68a1,1,0,0,0,1.42,0A1,1,0,0,0,21.71,20.29ZM11,18a7,7,0,1,1,7-7A7,7,0,0,1,11,18Z"
        />
      </svg>
    </div>
    <input
      class="emotion-3"
      id="search"
      name="search"
      placeholder="eg. Spider-Man"
      type="text"
    />
    <label
      for="search"
    >
      Search movies or TV series
    </label>
  </div>
  <div
    class="emotion-4"
  >
    <div
      class="flapper__Wrapper-kzbmuk-0 jYJyDw"
      size="78"
    >
      <div
        class="flapper__Ball-kzbmuk-1 gTEOpN"
        color="hsla(246, 88.5%, 62.4%, 1)"
        size="78"
        x="-26"
        y="13"
      />
      <div
        class="flapper__Ball-kzbmuk-1 cCTXgQ"
        color="hsla(246, 88.5%, 62.4%, 1)"
        size="78"
        x="4.491427816253161"
        y="-1.6838977275093896"
      />
      <div
        class="flapper__Ball-kzbmuk-1 lEfGR"
        color="hsla(246, 88.5%, 62.4%, 1)"
        size="78"
        x="12.022188575091121"
        y="-34.678316424296256"
      />
      <div
        class="flapper__Ball-kzbmuk-1 bgGXxN"
        color="hsla(246, 88.5%, 62.4%, 1)"
        size="78"
        x="-9.078534174415228"
        y="-61.13778584819434"
      />
      <div
        class="flapper__Ball-kzbmuk-1 imBqUM"
        color="hsla(246, 88.5%, 62.4%, 1)"
        size="78"
        x="-42.92146582558476"
        y="-61.13778584819435"
      />
      <div
        class="flapper__Ball-kzbmuk-1 eMxqxJ"
        color="hsla(246, 88.5%, 62.4%, 1)"
        size="78"
        x="-64.02218857509112"
        y="-34.678316424296234"
      />
      <div
        class="flapper__Ball-kzbmuk-1 eyqkaq"
        color="hsla(246, 88.5%, 62.4%, 1)"
        size="78"
        x="-56.491427816253164"
        y="-1.6838977275093967"
      />
    </div>
  </div>
  <h3>
    There are no results :(
  </h3>
</div>
`;
